x-pack/plugins/observability_solution/inventory/public/utils/extract_pagination_parameter.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isRight } from 'fp-ts/lib/Either';
+import { type EntityPagination, entityPaginationRt } from '../../common/entities';
+
+export function extractPaginationParameter(paginationParam?: string): EntityPagination | undefined {
+  if (paginationParam === undefined) {
+    return undefined;
+  }
+
+  const pagination = entityPaginationRt.decode(paginationParam);
+
+  if (isRight(pagination)) {
+    return pagination.right;
+  } else {
+    throw new Error('Invalid pagination object', { cause: pagination.left });
+  }
+}
```

Comment:
This [`decodeOrThrow` helper ](https://github.com/crespocarlos/kibana/blob/main/packages/kbn-io-ts-utils/src/decode_or_throw.ts), seems to do exactly what this function does.

Reply:
I have a feeling it will be a while before I find out all the little nice things hidden away in the codebase 😄